paymentsource.go

```diff
@@ -87,6 +87,7 @@ func SourceParamsFor(obj interface{}) (*SourceParams, error) {
 // The Type should indicate which object is fleshed out (eg. BitcoinReceiver or Card)
 // For more details see https://stripe.com/docs/api#retrieve_charge
 type PaymentSource struct {
+	APIResource
```

Comment:
In other languages this is not an API resource it's an interface. Curious how that fits with "only the top-level object has this set" and in which case that would be `PaymentSource` instead of `Card` for example that has `APIResource`

Reply:
For better or worse, we use `PaymentSource` exactly like an API resource. e.g.:

``` go
func (s Client) New(params *stripe.CustomerSourceParams) (*stripe.PaymentSource, error) {
	if params == nil {
		return nil, errors.New("params should not be nil")
	}

	if params.Customer == nil {
		return nil, errors.New("Invalid source params: customer needs to be set")
	}

	path := stripe.FormatURLPath("/v1/customers/%s/sources", stripe.StringValue(params.Customer))
	source := &stripe.PaymentSource{}
	err := s.B.Call(http.MethodPost, path, s.Key, params, source)
	return source, err
}
```

Reply:
Yeah but my question was more about the limitation where only the top-level object has `APIResource` fields. So for example if you list Invoices, the List has the info but each Invoice (or invoice line items) doesn't. 

I think that is fine but in the case of PaymentSource it is a bit stranger because the real API resource you are fetching is the Card, but the part that has `APIResource` filled would be `PaymentSource` and never `Card` right?

Reply:
Ah, I see.

> I think that is fine but in the case of PaymentSource it is a bit stranger because the real API resource you are fetching is the Card, but the part that has `APIResource` filled would be `PaymentSource` and never `Card` right?

Yeah, maybe. I think though in this case it's still okay because even if it's a little surprising if you know the API's domain very profoundly, it's not if you're just familiar with the stripe-go conventions. Basically:

1. The immediate object returned from a call like `coupon.*` or `paymentsource.*` will always have the `APIResource` fields set.
2. No other objects, including subobjects, will have the `APIResource` fields set. If you need them, you'll have to get them from the parent.

If you have an idea of an alternative that's more sane, I'm open to it, but personally I think it's okay like this.

And one last thought: if this turns out to be a bigger problem than I expect, we can always address this later by doing a recursive population of `APIResource`. It's easier to transition from nil to non-nil in those fields with a later change than it would be to go the other direction.